### PR TITLE
source resolver should only react to sources

### LINKED ISSUE #1488: constantly updating sources
I've noticed that if the source watcher is enabled the log spins on `updating sources`. I noticed this when using it with an sbt project using Java 7.

I wonder if the source watcher code should just be removed and made a client responsibility... maybe we make it false by default.

### PR DESCRIPTION
**Pull request by fommil**
close #1488

### CODE REVIEW

**Comment by fommil** [comment]
I ported the presentation compiler crash fix too.

File: core/src/main/scala/org/ensime/indexer/SourceResolver.scala [hunk 1]
```diff
@@ -19,12 +21,15 @@ class SourceResolver(
...
-  // it's not worth doing incremental updates - this is cheap
-  // (but it would be nice to have a "debounce" throttler)
-  def fileAdded(f: FileObject) = update()
-  def fileRemoved(f: FileObject) = update()
+  def fileAdded(f: FileObject) = if (relevant(f)) update()
+  def fileRemoved(f: FileObject) = if (relevant(f)) update()
   def fileChanged(f: FileObject) = {}
 
+  def relevant(f: FileObject): Boolean = f.getName.isFile && {
+    val file = new File(f.getName.getURI)
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/indexer/SourceResolver.scala]: Could we avoid the new File creation here - not sure how often this is called/whether it hurts performance at all - but thought I would mention it.

Comment (rorygraves) [inline core/src/main/scala/org/ensime/indexer/SourceResolver.scala]: @fommil  says 'It's only called for changed files so it's not a churn hotspot'
LGTM

**Comment by rorygraves** [comment]
LGTM

**Pull request merged**